src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -848,6 +830,55 @@ private static int GetAlgorithmNid(HashAlgorithmName hashAlgorithmName)
             return nid;
         }
 
+        private static void ValidatePadding(RSAEncryptionPadding padding)
+        {
+            if (padding == null)
+            {
+                throw new ArgumentNullException(nameof(padding));
+            }
+
+            if (padding.Mode == RSAEncryptionPaddingMode.Pkcs1)
+            {
+                // Fail if any options are set other than the mode.
+                if (padding != RSAEncryptionPadding.Pkcs1)
+                {
+                    throw PaddingModeNotSupported();
+                }
+            }
+            else if (padding.Mode != RSAEncryptionPaddingMode.Oaep)
+            {
+                throw PaddingModeNotSupported();
+            }
```

Comment:
```suggestion
            if (padding.Mode != RSAEncryptionPaddingMode.Oaep &&
                (padding.Mode != RSAEncryptionPaddingMode.Pkcs1 || padding != RSAEncryptionPadding.Pkcs1)
            {
                    throw PaddingModeNotSupported();
            }
```
?

Reply:
Went a step further and eliminated the redundant check of the Mode value:

```C#
            if (padding.Mode != RSAEncryptionPaddingMode.Oaep &&
                padding != RSAEncryptionPadding.Pkcs1)
            {
```

Required a big comment to explain the asymmetry, but I think it ends up looking clearer than the mixed ands and ors :smile:.